### ISSUE TITLE
Implement mounted host folder support using Windows shares

### ISSUE THREAD

**Issue by hferentschik** [minishift/minishift#316]
Following up on the investigation host folder mounts on Windows + Hyper-V (issue #293)

**Comment by gbraad** [comment]
Hopefully this can all be automated. On the host you can create a shared folder using PowerShell:

```
Set-Variable MINISHIFTHOME $HOME\.minishift
Set-Variale MINISHIFTSHARED $MINISHIFTHOME\Shared

if (!(Test-Path $MINISHIFTSHARED )) { 
   New-Item MINISHIFTSHARED -type Directory 
}

New-SMBShare –Name "MinishitShare" –Path MINISHIFTSHARED `
   –ContinuouslyAvailable $True `
   –FullAccess mygroup\administrators  `
   -ChangeAccess "mygroup\authenticated users" `
   -ReadAccess "mygroup\authenticated users"
```

and then inside the VM this share can be accessed and mounted. Note that you might have to set to EVERYONE, but this is not very safe.

**Comment by hferentschik** [comment]
@gbraad, nice. So, you are saying this is completely password free? Anyone can mount this Samba share and read/write to it? I think when I did some manual tests, I mounted the share using the users credentials? Obviously password free would be easier for us, question is whether this is not a bit too open? What does Docker for Windows do?

Also, what's about Windows 7? How do you do it there w/o PowerShell. My understanding is that most/many Windows developers are still on this OS version and we need to cater for them as well.

**Comment by gbraad** [comment]
@hferentschik Windows7 has no Hyper-V option. It is only available from 2012, 8.x and up. For Windows7 you have to use VirtualBox.

The password is actually an issue. You can use "Authenticated Users", but this needs to be an existing user. We can also create a user for use with PowerShell. How will we deal with the password? Generate one with a random value... and then remove the user on `minishift delete`?

There is another issue with the share, how long will it exist? `new-smbshare` has an option: `-ContinuouslyAvailable<Boolean>` [TechNet][technet continuouslyavailable]

[technet continuouslyavailable]: https://technet.microsoft.com/en-us/library/jj635722.aspx

**Comment by hferentschik** [comment]
> Windows7 has no Hyper-V option. 

Of course, I forgot. Windows 7 is VirtualBox only. However, given that we don't know whether we can use guest additions, the share options is a viable solution there as well, right? 

> The password is actually an issue. You can use "Authenticated Users", but this needs to be an existing user. We can also create a user for use with PowerShell. How will we deal with the password? 

I would have thought that we use the users password which then gets stored in the VM. Obviously this is a bit of a security issue. What does Docker for Windows do?

> There is another issue with the share, how long will it exist?

Easiest of course would be to just keep it. If we can tie it to the lifecycle, it would need to be deleted on 'minishift delete'. Can one share the same drive/path multiple times? Say a user already shares his home directory for other purposes. Can we create a new share or should we reuse the existing one. In the latter case, we cannot just delete it.

**Comment by gbraad** [comment]
> What does Docker for Windows do?

It is described here: https://docs.docker.com/docker-for-windows/#/shared-drives Will look if I can find something more about the implementation

**Comment by hferentschik** [comment]
> You will be asked to provide your Windows system username and password (domain user) to apply shared drives. You can select an option to have Docker store the credentials so that you don’t have to re-enter them every time.

That seems reasonable

**Comment by gbraad** [comment]
Note: shares can also be created with the old `NET SHARE` command [TechNet](https://technet.microsoft.com/en-us/library/bb490712.aspx).

**Comment by gbraad** [comment]
Added `cifs-utils` to the centos-iso: https://github.com/minishift/minishift-centos-iso/issues/98

**Comment by gbraad** [comment]
Added `cifs-utils` and dependencies to the b2d-iso: minishift/minishift-b2d-iso#30

**Comment by gbraad** [comment]
From the command line I am able to mount the Users shared folder on Windows 10:
```
$ minishift.exe ssh "sudo mkdir -p /mnt/sda1/users"
$ minishift.exe ssh "sudo mount -t cifs //10.0.21.33/Users /mnt/sda1/users -o username=[local username],password=[local password]"
$ minishift.exe ssh "ls -al /mnt/sda1/users/[username]"
```
However, still experiencing issues with #422.

**Comment by budhrg** [comment]
IMO, we should document this case also https://github.com/minishift/minishift/issues/422#issuecomment-281009269

where user users  Microsoft Account.

**Comment by gbraad** [comment]
At the moment I am on documenting this. As for a MicrosoftAccount is is necessary to specify the domain=[local computername] when mounting. Would be ideal to automate this, but will have to look how we can get the following information with the Go api...

**Comment by hferentschik** [comment]
Merged via pull request #449
